sfputil/debug.py

```diff
@@ -0,0 +1,327 @@
+import sys
+import click
+import sonic_platform
+import sonic_platform_base.sonic_sfp.sfputilhelper
+from sonic_platform_base.sfp_base import SfpBase
+from sonic_py_common import multi_asic
+import utilities_common.cli as clicommon
+from utilities_common import platform_sfputil_helper
+from utilities_common.platform_sfputil_helper import (
+    logical_port_name_to_physical_port_list,
+    logical_port_to_physical_port_index,
+    is_rj45_port,
+    is_sfp_present,
+)
+from swsscommon.swsscommon import SonicV2Connector, ConfigDBConnector
+
+# Global variables for platform-specific utilities
+platform_sfputil = None
+platform_chassis = None
+
+EXIT_FAIL = 1  # Exit code for failure
+ERROR_NOT_IMPLEMENTED = 2  # Error code for unimplemented functionality
+
+
+def get_subport_lane_mask(subport, lane_count):
+    """
+    Get the lane mask for the given subport and lane count.
+
+    This method calculates the lane mask based on the subport and lane count.
+
+    Args:
+        subport (int): The subport number to calculate the lane mask for.
+        lane_count (int): The number of lanes per subport.
+
+    Returns:
+        int: The lane mask calculated for the given subport and lane count.
+    """
+    # Calculating the lane mask using bitwise operations.
+    return ((1 << lane_count) - 1) << ((subport - 1) * lane_count)
+
+
+def get_sfp_object(port_name):
+    """
+    Retrieve the SFP object for a given port.
+
+    This function checks whether the port is a valid RJ45 port or if an SFP is present.
+    If valid, it retrieves the SFP object for further operations.
+
+    Args:
+        port_name (str): The name of the logical port to fetch the SFP object for.
+
+    Returns:
+        SfpBase: The SFP object associated with the port.
+
+    Raises:
+        SystemExit: If the port is an RJ45 or the SFP EEPROM is not present.
+    """
+    # Retrieve the physical port corresponding to the logical port.
+    physical_port = logical_port_to_physical_port_index(port_name)
+    # Fetch the SFP object for the physical port.
+    sfp = platform_chassis.get_sfp(physical_port)
+
+    # Check if the port is an RJ45 port and exit if so.
+    if is_rj45_port(port_name):
+        click.echo(f"{port_name}: This functionality is not applicable for RJ45 port")
+        sys.exit(EXIT_FAIL)
+
+    # Check if the SFP EEPROM is present and exit if not.
+    if not is_sfp_present(port_name):
+        click.echo(f"{port_name}: SFP EEPROM not detected")
+        sys.exit(EXIT_FAIL)
+
+    return sfp
+
+
+def get_subport(port_name):
```

Comment:
@vdahiya12 should this be common util?

Reply:
@vdahiya12 lets make this generic such a way that I can read any field from the Port table? Have this in the common place?

Reply:
today its subport, tomorrow there will be another field to query we will keep adding new fn

Reply:
fixed